samcli/commands/init/init_templates.py

```diff
@@ -112,14 +105,23 @@ def _check_app_template(entry: Dict, app_template: str) -> bool:
         return bool(entry["appTemplate"] == app_template)
 
     def init_options(self, package_type, runtime, base_image, dependency_manager):
-        if not self.clone_attempted:
-            self._clone_repo()
-        if self.repo_path is None:
+        if not self._git_repo.clone_attempted:
```

Comment:
Looks like `clone_attempted` is never used inside `GitRepo`, I think it should be moved out and stored here instead